tests/test_atom_mapping.py

```diff
@@ -953,3 +971,281 @@ def test_get_cores_and_diagnostics():
+
+
+def test_initial_mapping_returns_self_with_same_params(hif2a_ligands):
+    """If an initial map is provided for the same parameters, the values are identical"""
+    rng = np.random.default_rng(2024)
+    mol_a, mol_b = rng.choice(hif2a_ligands, 2)
+
+    kwargs = DEFAULT_ATOM_MAPPING_KWARGS.copy()
+    cores = atom_mapping.get_cores(mol_a, mol_b, **kwargs)
+
+    kwargs["initial_mapping"] = cores[0]
+
+    # Since the initial mapping is set and no parameters are changed, should only return a single core
+    identity_core = atom_mapping.get_cores(mol_a, mol_b, **kwargs)
+
+    assert len(cores) > 1
+    assert len(identity_core) == 1
+    # Core ordering does get shuffled, but pairs should be identical
+    assert set(tuple(pair) for pair in identity_core[0]) == set(tuple(pair) for pair in cores[0])
+
+
+def test_initial_mapping_always_a_subset_of_cores(hif2a_ligands):
+    rng = np.random.default_rng(2024)
+    mols = hif2a_ligands
+    mol_a, mol_b = rng.choice(mols, 2)
+
+    kwargs = DEFAULT_ATOM_MAPPING_KWARGS.copy()
+    cores = atom_mapping.get_cores(mol_a, mol_b, **kwargs)
+
+    initial_map = cores[0]
+
+    kwargs["initial_mapping"] = initial_map
+    kwargs["ring_cutoff"] = kwargs["ring_cutoff"] * 2.0
+    kwargs["chain_cutoff"] = kwargs["chain_cutoff"] * 2.0
+
+    cores_with_map = atom_mapping.get_cores(mol_a, mol_b, **kwargs)
+
+    initial_pairs = set(tuple(pair) for pair in initial_map)
+    for core in cores_with_map:
+        new_pairs = set(tuple(pair) for pair in core)
+        assert new_pairs.issuperset(initial_pairs)
+
+
+@pytest.mark.parametrize("param_to_change", ["ring_matches_ring_only", "connected_core"])
+def test_initial_mapping_ignores_filters(hif2a_ligands, param_to_change):
+    mol_a = hif2a_ligands[0]
+    mol_b = hif2a_ligands[1]
+    cores = atom_mapping.get_cores(mol_a, mol_b, **DEFAULT_ATOM_MAPPING_KWARGS)
+
+    assert DEFAULT_ATOM_MAPPING_KWARGS[param_to_change]
+
+    kwargs = DEFAULT_ATOM_MAPPING_KWARGS.copy()
+    kwargs[param_to_change] = False
+    cores_without_ring_match_ring = atom_mapping.get_cores(mol_a, mol_b, **kwargs)
+    # The core without the ring matches ring is larger
+    assert len(cores[0]) < len(cores_without_ring_match_ring[0])
+
+    initial_map_kwargs = DEFAULT_ATOM_MAPPING_KWARGS.copy()
+    initial_map_kwargs["initial_mapping"] = cores_without_ring_match_ring[0]
+
+    if param_to_change == "ring_matches_ring_only":
+        # If we remap with this core that is invalid under the mapping conditions, return the original core
+        new_cores = atom_mapping.get_cores(mol_a, mol_b, **initial_map_kwargs)
+        assert len(new_cores) == 1
+    else:
+        # If there is no connected core that can be made from the disconnected will raise an exception
+        with pytest.raises(NoMappingError):
+            atom_mapping.get_cores(mol_a, mol_b, **initial_map_kwargs)
+
+
+def test_hybrid_core_generation(hif2a_ligands):
+    """
+    Verify that generating"""
```

Comment:
possibly remove docstring

Reply:
Woops starting writing this docstring then left it lo. Will fix it up.

Reply:
Completed writing it in https://github.com/proteneer/timemachine/pull/1254/commits/8be9c2143a0a11522de1bbafbfb1b603d8b71390